멀티모듈 프로젝트 구조 전환 (initial)

xenex90 opened the pull request:
멀티모듈 프로젝트 구조 전환 (initial)
@f-lab-lyan 멘토님,

기존 프로젝트 삭제 후 기존 프로젝트 컨셉 유지한 개발 초기 셋팅입니다.
아래는 PR 요청 요약입니다.

- Spring Boot 3.5.0, Java 21 환경 설정
- user-service, content-service, common, database 등 여러 모듈을 가정
- 빌드 성공 확인 (test 제외)

xenex90 committed:
멀티모듈 프로젝트 구조 전환

- 모놀리틸에서 멀티모듈 아키텍처로 변경
- user-service: 사용자 관리 및 인증 기능
- content-service: 콘텐츠 생성 및 관리 기능
- community-service: 커뮤니티 기능 및 상호작용
- code-review-service: 코드 리뷰 기능
- common: 공통 유틸리티 및 DTO 모듈
- database: 엔티티 및 리포지토리 계층
- Gradle 멀티모듈 빌드 시스템 구성
- Spring Boot 3.5.0, Java 21 환경 설정"

f-lab-lyan commented on everybodycodeacademy/build.gradle.kts:
```diff
@@ -0,0 +1,54 @@
+}
+
+subprojects {
+    apply(plugin = "java")
```
여기에 subprojects로 설정이 있는데, 실제로 subprojects에서도 동일한 설정을 하고 있는 것 같아요. 둘 중에 하나는 없어도 되지 않을까요? 그리고 gradle document에 보면, 이제 subprojects는 사용하지 말라는 안내가 있기도 하죠. 😀

xenex90 replied:
subprojects의 대안 방법을 몰라서 buildSrc폴더에 build.gradle.kts, src/main/kotiln/convention-java.gradle.kts, src/main/kotiln/convention-spring-boot.gradle.kts을 추가를 우선 진행했는데 여기까진 빌드 됩니다.


subprojects 코드부분은 곧 삭제 예정입니다. 삭제 이후에는 끝인지 아닌지 모르지만 좀 더 진행해보겠습니다.

f-lab-lyan commented on everybodycodeacademy/code-review-service/src/main/java/f_lab/codereviewservice/CodeReviewServiceApplication.java:
```diff
@@ -1,13 +1,13 @@
-package com.every.everycodeacademy;
+package f_lab.codereviewservice;
```
일반적으로 _는 잘 사용하지 않는 것 같아요.

xenex90 replied:
넵 로컬에서 flab으로 수정했습니다

f-lab-lyan commented on everybodycodeacademy/common/src/main/java/f_lab/common/CommonApplication.java:
```diff
@@ -1,13 +1,13 @@
 
     public static void main(String[] args) {
-        SpringApplication.run(LoginProjectSpringBootApplication.class, args);
+        SpringApplication.run(CommonApplication.class, args);
```
common도 별도의 어플리케이션으로 뜨는 건가요?

xenex90 replied:
로컬에서 공통 모듈로 변경하였습니다.

f-lab-lyan reviewed (commented):
코드를 살펴 봤는데 필요없는 파일들이 있는 것 같아요. 스프링 엎,ㄹ리케이션 메인 파일들도 필요한 곳에만 넣으면 좋을 것 같아요. 그리고, 멀티 모듈 설정도 필요한 설정만 남기면 좋을 것 같습니다.

f-lab-lyan commented:
https://docs.gradle.org/current/userguide/sharing_build_logic_between_subprojects.html#using_a_composite_build_named_build_logic

f-lab-lyan commented:
전체 디렉토리 구조를 한 단계 위로 올려야 할 것 같아요.

xenex90 commented:
> 전체 디렉토리 구조를 한 단계 위로 올려야 할 것 같아요.

넵~

xenex90 committed:
chore: 프로젝트 구조 리팩토링 및 Gradle convention 플러그인 적용

- 기존 모듈별/루트 Gradle 설정(build.gradle.kts, settings.gradle.kts, wrapper 등)과 application.properties 삭제
- buildSrc에 convention-java, convention-spring-boot 플러그인 추가
- 각 서브모듈은 convention 플러그인만 적용하도록 build.gradle.kts 재구성
- 공통 설정(자바 툴체인, 리포지토리, BOM 등)을 중앙화하여 중복 제거

xenex90 commented:
@f-lab-lyan 멘토님

말씀하신 것 참고해서 수정해서 다시 커밋, 푸시 하였습니다~
[chore: 프로젝트 구조 리팩토링 및 Gradle convention 플러그인 적용](https://github.com/f-lab-edu/everybodycodeacademy/pull/28/commits/d31e23c15a54f21bea867fd30819b10a47d9c8f7)

xenex90 committed:
불필요한 주석 제거

xenex90 committed:
Delete everybodycodeacademy/gradlew.bat

xenex90 committed:
Delete everybodycodeacademy/gradle/wrapper/gradle-wrapper.properties

불필요한 파일 제거

xenex90 committed:
Delete everybodycodeacademy/gradlew

불필요한 파일 제거

xenex90 reviewed (commented):
파일 확인하였습니다.

f-lab-lyan reviewed (approved):
*-service 들은 다 다른 인스턴스로 쓰는 것인가요? 그런 의도라면 괜찮은 것 같습니다. 수고하셨습니다. 👍

xenex90 commented:
> *-service 들은 다 다른 인스턴스로 쓰는 것인가요? 그런 의도라면 괜찮은 것 같습니다. 수고하셨습니다. 👍

넵 각각입니다.
